apiserver/facades/agent/upgradesteps/upgradesteps.go

```diff
@@ -97,6 +126,45 @@ func (api *UpgradeStepsAPI) ResetKVMMachineModificationStatusIdle(arg params.Ent
 	return result, nil
 }
 
+// WriteUniterState did not exist prior to v2.
+func (*UpgradeStepsAPIV1) WriteUniterState(_, _ struct{}) {}
+
+// WriteUniterState write the uniter state for the set of units provided.
+// Related to the move of the uniter state from a local file to the controller.
+func (api *UpgradeStepsAPI) WriteUniterState(args params.SetUnitStateArgs) (params.ErrorResults, error) {
```

Comment:
Is it possible to delegate this operation to the uniter facade or would that be too much effort? The implementation looks the same

Reply:
Worker's should only share facades if they are in apiserver/.../common. Then we'd have the uniter state operations split between common and the uniter, as upgrade steps won't use the State() piece.  It'd also be a bit weird as the uniter and upgradsteps would look very different on the api/common side.  Would we only implement the apiserver common piece? We'd also have to guard against mucking up the upgrade steps for 2.8 if this changes in the future on the uniter side.

Reply:
Makes sense. Thanks for the clarification.